src/main/java/com/develetter/develetter/mail/batch/MailBatch.java

```diff
@@ -0,0 +1,141 @@
+import com.develetter.develetter.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.batch.core.*;
+import org.springframework.batch.core.job.builder.JobBuilder;
+import org.springframework.batch.core.repository.JobRepository;
+import org.springframework.batch.core.step.builder.StepBuilder;
+import org.springframework.batch.item.ItemProcessor;
+import org.springframework.batch.item.data.RepositoryItemReader;
+import org.springframework.batch.item.data.RepositoryItemWriter;
+import org.springframework.batch.item.data.builder.RepositoryItemReaderBuilder;
+import org.springframework.batch.item.data.builder.RepositoryItemWriterBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.domain.Sort;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.thymeleaf.spring6.SpringTemplateEngine;
+
+import java.util.Map;
+
+@Configuration
+@RequiredArgsConstructor
+@Slf4j
+public class MailBatch {
+    private final JavaMailSender javaMailSender;
+    private final SpringTemplateEngine templateEngine;
+
+    private final JobRepository jobRepository;
+    private final MailRepository mailRepository;
+    private final UserRepository userRepository;
+    private final PlatformTransactionManager platformTransactionManager;
+
+    private final ConferenceCalendarService conferenceCalendarService;
+    private final AsyncMailService asyncMailService;
+    private final MailService mailService;
+
+
+    private static final int CHUNK_SIZE = 5;
+
+    @Bean
+    public Job mailJob() {
+        return new JobBuilder("mailJob", jobRepository)
+                .start(saveMailStep())
+                .next(sendMailStep())
+                .build();
+    }
+
+    //메일 내용 저장
+    @Bean
+    public Step saveMailStep() {
+        return new StepBuilder("saveMailStep", jobRepository)
+                .<UserEntity, Mail> chunk(CHUNK_SIZE, platformTransactionManager)
+                .reader(userReader())
+                .processor(saveMailProcessor())
+                .writer(mailWriter())
+                .build();
+    }
+
+    @Bean
+    public RepositoryItemReader<UserEntity> userReader() {
+        return new RepositoryItemReaderBuilder<UserEntity>()
+                .name("userReader")
+                .pageSize(CHUNK_SIZE)
+                .methodName("findAll")
+                .repository(userRepository)
+                .sorts((getSortMap()))
+                .build();
+    }
+
+    @Bean
+    public ItemProcessor<UserEntity, Mail> saveMailProcessor() {
+        return new ItemProcessor<UserEntity, Mail>() {
+            @Override
+            public Mail process(UserEntity user) throws Exception {
+                return new Mail(user.getId());
+            }
+        };
+    }
```

Comment:
```suggestion
           return user -> new Mail(user.getId());
```

이렇게 수정하거나, 여기에 메일 저장 로직이 포함되면 좋을 것 같아요.

Reply:
수정했습니다~